cmd/influxd/launcher/cmd.go

```diff
@@ -66,26 +69,45 @@ func setCmdDescriptions(cmd *cobra.Command) {
 `
 }
 
-func cmdRunE(ctx context.Context, o *InfluxdOpts) func() error {
+func cmdRunE(o *InfluxdOpts) func() error {
 	return func() error {
 		fluxinit.FluxInit()
+		l := NewLauncher()
 
-		// exit with SIGINT and SIGTERM
-		ctx = signals.WithStandardSignals(ctx)
+		// Set up logging to STDOUT.
+		logconf := &influxlogger.Config{
+			Format: "auto",
+			Level:  o.LogLevel,
+		}
+		logger, err := logconf.New(os.Stdout)
+		if err != nil {
+			return err
+		}
 
-		l := NewLauncher()
-		if err := l.run(ctx, o); err != nil {
+		if err := l.runUntilCanceled(o, logger); err != nil {
 			return err
 		}
-		<-ctx.Done()
+		l.gracefulShutdown(logger)
 
-		// Attempt clean shutdown.
-		ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
-		defer cancel()
-		l.Shutdown(ctx)
+		return logger.Sync()
+	}
+}
 
-		return nil
+// runUntilCanceled runs the given launcher and waits, only exiting when a SIGINT or SIGTERM is received.
+func (l *Launcher) runUntilCanceled(o *InfluxdOpts, log *zap.Logger) error {
+	ctx := signals.WithStandardSignals(context.Background())
```

Comment:
should probably create this context value before calling `cmdRunE()` and pass that value up the call chain.